packages/react-router-dom/index.tsx

```diff
@@ -976,6 +977,95 @@ export function useFormAction(
   return createPath(path);
 }
 
+let blockerKey = "blocker-singleton";
```

Comment:
I think a singleton will fail in many cases, right? Like if I use useBlocker or usePrompt in two places in the application, won't the second one override the settings of the first?

Reply:
This is correct. We are planning to tackle multiple blockers separately, after the initial phase is done and we've worked through all of quirks.